src/sqlfluff/core/templaters/jinja.py

```diff
@@ -200,6 +204,152 @@ def __str__(self):  # pragma: no cover TODO?
         }
         return dbt_builtins
 
+    @staticmethod
+    def _generate_airflow_builtins():
+        """Generate the airflow builtins which are injected in the context.
+
+        https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html
+        """
+        # Copied from _generate_dbt_builtins
+        # This feels a bit wrong defining these here, they should probably
+        # be configurable somewhere sensible. But for now they're not.
+        # TODO: Come up with a better solution.
+
+        class DagEmulator:
+            """A class which emulates the `DAG` class from Airflow."""
+
+            dag_id = "sample_dag"
+            is_subdag = False
+
+            def __str__(self):
+                return self.dag_id
+
+        class DagRunEmulator:
+            """A class which emulates the `task` class from Airflow."""
+
+            dag_id = "sample_dag"
+            run_id = "sample_dag__run_id"
+            execution_date = datetime.datetime.now(datetime.timezone.utc)
```

Comment:
Will SQLFluff users want or need to configure the value of individual object fields, both these dynamic ones as well as ones that you've hardcoded?

Reply:
Of course it's thinkable that users have some very specific DAG setup, and do some filtering within Jinja on the `dag_id`s. However, I have rarely seen these being used in SQL in the first place. Most of the time they are used in INSERT statements or such.

My vision here is: I want to provide some builtins that covers "normal" use of the Airflow builtins. When user do something super crazy with the Airflow variables on template level, they can (and should) always fallback to providing them themselves via `.sqlfluff` config.